twoliter/build.rs

```diff
@@ -11,56 +11,108 @@ use bytes::BufMut;
 use flate2::write::ZlibEncoder;
 use flate2::Compression;
 use std::path::{Path, PathBuf};
+use std::time::SystemTime;
 use std::{env, fs};
 
 const DATA_INPUT_DIR: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/embedded");
+const TOOLS_SOURCECODE_DIR: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/../tools");
 
 fn main() {
-    println!("cargo:rerun-if-changed={}", DATA_INPUT_DIR);
+    let paths = Paths::new();
+    println!("cargo:rerun-if-changed={}", paths.data_input_dir.display());
+    println!(
+        "cargo:rerun-if-changed={}",
+        paths.tools_sourcecode_dir.display()
+    );
     // Make sure we run again if the target triple (i.e. aarch64-unknown-linux-gnu) changes.
     println!("cargo:rerun-if-env-changed=TARGET");
-    let data_input_dir = PathBuf::from(DATA_INPUT_DIR);
-
-    // This is the directory that cargo creates for us so that we can pass things from the build
-    // script to the main compilation phase.
-    let out_dir =
-        PathBuf::from(env::var("OUT_DIR").expect("The cargo variable 'OUT_DIR' is missing"));
 
-    // This is where we will copy all of the things we want to add to our tarball. We will then
-    // compress and tar this directory.
-    let tools_dir = out_dir.join("tools");
-    fs::create_dir_all(&tools_dir).expect(&format!(
+    fs::create_dir_all(&paths.prep_dir).expect(&format!(
         "Unable to create directory '{}'",
-        tools_dir.display()
+        paths.prep_dir.display()
     ));
 
-    // This is the filepath to the tarball we will create.
-    let tar_path = out_dir.join("tools.tar.gz");
+    paths.copy_file("Dockerfile");
+    paths.copy_file("Makefile.toml");
+    paths.copy_file("docker-go");
+    paths.copy_file("partyplanner");
+    paths.copy_file("rpm2img");
+    paths.copy_file("rpm2kmodkit");
+    paths.copy_file("rpm2migrations");
```

Comment:
Do we need to emit `rerun-if-changed` for the non-Rust binary artifacts?

Reply:
> Do we need to emit `rerun-if-changed` for the non-Rust binary artifacts?

I am! `println!("cargo:rerun-if-changed={}", paths.data_input_dir.display());`